pallet/liquidity-pools/src/lib.rs

```diff
@@ -0,0 +1,1065 @@
+				false,
+			)?;
+
+			Self::deposit_event(Event::PoolOpen {
+				pool_id: id,
+				asset_id,
+				interest_rate,
+				max_tokens,
+				lock_start_block,
+				lock_end_block,
+			});
+			Ok(())
+		}
+
+		/// Sets up a successor relationship between two pools.
+		///
+		/// This function allows admin users to link pools in a sequential manner, so when one pool
+		/// ends, users who are set to rollover can automatically join the successor pool to
+		/// continue receiving rewards.
+		///
+		/// Parameters:
+		/// - `origin`: The origin account setting the pool succession. Must be an admin.
+		/// - `predecessor_pool_id`: The ID of the predecessor pool that will link to a successor
+		///   upon completion.
+		/// - `successor_pool_id`: The ID of the successor pool that will continue from where the
+		///   predecessor pool ends.
+		///
+		/// Restrictions:
+		/// - Both the predecessor and successor pools must exist.
+		/// - The `max_tokens` of the successor pool should be greater than or equal to the
+		///   predecessor pool.
+		///
+		/// Emits `SetSuccession` event when successful.
+		#[pallet::weight(T::WeightInfo::set_pool_succession())]
+		pub fn set_pool_succession(
+			origin: OriginFor<T>,
+			predecessor_pool_id: T::PoolId,
+			successor_pool_id: T::PoolId,
+		) -> DispatchResult {
+			T::ApproveOrigin::ensure_origin(origin)?;
+
+			// Check that predecessor exists
+			let predecessor_pool = Pools::<T>::get(predecessor_pool_id)
+				.ok_or(Error::<T>::PredecessorPoolDoesNotExist)?;
+			// Check that successor exists
+			let successor_pool =
+				Pools::<T>::get(successor_pool_id).ok_or(Error::<T>::SuccessorPoolDoesNotExist)?;
+
+			// Check successor max_tokens is greater than predecessor max_tokens
+			ensure!(
+				successor_pool.max_tokens >= predecessor_pool.max_tokens,
+				Error::<T>::SuccessorPoolSizeShouldBeGreaterThanPredecessor
+			);
+			ensure!(
+				successor_pool.lock_start_block > predecessor_pool.lock_end_block,
+				Error::<T>::SuccessorPoolSizeShouldBeLockedAfterPredecessor
+			);
+
+			<PoolRelationships<T>>::insert(
+				&predecessor_pool_id,
+				PoolRelationship { successor_id: Some(successor_pool_id) },
+			);
+
+			Self::deposit_event(Event::SetSuccession { predecessor_pool_id, successor_pool_id });
+
+			Ok(())
+		}
+
+		/// Updates a user's preference for rolling over to the next pool.
+		///
+		/// Users can choose to automatically transfer their stake to a successor pool
+		/// once the current pool ends, allowing for continuous reward accumulation without manual
+		/// intervention.
+		///
+		/// Parameters:
+		/// - `origin`: The account of the user setting the rollover preference.
+		/// - `id`: The ID of the pool for which the rollover preference is being set.
+		/// - `should_roll_over`: A boolean indicating whether the user's stake should rollover to a
+		///   successor pool.
+		///
+		/// Restrictions:
+		/// - The pool must be in the `Provisioning` status, not yet active or closed.
```

Comment:
```suggestion
		/// - The pool must be in the `Open` status, not yet active or closed.
```

nit: I personally do think that `Provisioning` and `Provisioned` are more relevant than open/closed (but thats personal taste)